engine/src/main/java/org/terasology/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -120,7 +119,11 @@ private void generateIntermediateList(List<Node> orderedNodes) {
             intermediateList.add(node);
             // Add state changes to reset all desired state changes back to default.
             for (StateChange stateChange : node.getDesiredStateChanges()) {
-                intermediateList.add(stateChange.getDefaultInstance());
+                // TODO: consider using optional here?
+                StateChange defaultInstance = stateChange.getDefaultInstance();
+                if (defaultInstance != null) {
```

Comment:
I'm assuming later in this PR there is a _GOOD_ reason for a default instance to _EVER_ be null. :smiling_imp: